tool/tsh/proxy.go

```diff
@@ -383,22 +412,44 @@ func onProxyCommandDB(cf *CLIConf) error {
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		cmd, err := dbcmd.NewCmdBuilder(client, profile, routeToDatabase, rootCluster,
+		cmdMap, err := dbcmd.NewCmdBuilder(client, profile, routeToDatabase, rootCluster,
 			dbcmd.WithLocalProxy("localhost", addr.Port(0), ""),
 			dbcmd.WithNoTLS(),
 			dbcmd.WithLogger(log),
 			dbcmd.WithPrintFormat(),
-		).GetConnectCommand()
+		).GetConnectCommandOptions()
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		err = dbProxyAuthTpl.Execute(os.Stdout, map[string]string{
+
+		// shared template arguments
+		templateArgs := map[string]any{
 			"database": routeToDatabase.ServiceName,
 			"type":     dbProtocolToText(routeToDatabase.Protocol),
 			"cluster":  client.SiteName,
-			"command":  fmt.Sprintf("%s %s", strings.Join(cmd.Env, " "), cmd.String()),
 			"address":  listener.Addr().String(),
-		})
+		}
+
+		// format the command line
+		fmtMap := map[string]string{}
+		for key, cmd := range cmdMap {
+			fmtMap[key] = formatCommand(cmd)
+		}
+
+		// there is only one command, use plain template.
+		if len(fmtMap) == 1 {
+			var cmd string
+			for _, c := range fmtMap {
+				cmd = c
+			}
+			templateArgs["command"] = cmd
+			err = dbProxyAuthTpl.Execute(os.Stdout, templateArgs)
+			return trace.Wrap(err)
+		}
+
+		// multiple command options, use a different template.
+		templateArgs["commands"] = fmtMap
```

Comment:
Can we extract here into a separate function that returns the tmp and tmpArgs ?

Reply:
I don't follow - do you mean template and templateArgs?

Reply:
Yes the function can returned templateArgs and template or even build and execute the command template and return an error. I think that just extracting this inline logic from  `onProxyCommandDB`  function will improve readability.

Reply:
Done.